Analysis/src/TauUncertainties.cpp

```diff
@@ -107,90 +106,78 @@ double TauESUncertainties::GetCorrectionFactorTrueTau(analysis::Period period, i
+                                     {11, StVariable(0.0, 0.0, 0.0)}    }},
+      { analysis::Period::Run2017, { {0,  StVariable(0.989, 0.145, 0.137)},
+                                     {1,  StVariable(0.752, 0.141, 0.139)},
+                                     {5,  StVariable(0.0, 0.0, 0.0)},
+                                     {6,  StVariable(0.0, 0.0, 0.0)},
+                                     {10, StVariable(0.0, 0.0, 0.0)},
+                                     {11, StVariable(0.0, 0.0, 0.0)}    }},
+      { analysis::Period::Run2018, { {0,  StVariable(0.815, 0.106, 0.1)},
+                                     {1,  StVariable(0.62, 0.151, 0.156)},
+                                     {5,  StVariable(0.0, 0.0, 0.0)},
+                                     {6,  StVariable(0.0, 0.0, 0.0)},
+                                     {10, StVariable(0.0, 0.0, 0.0)},
+                                     {11, StVariable(0.0, 0.0, 0.0)}    }}
+    };
+
+    if(!deep_tau_vs_e_energy_scale_endcap.count(period) || !deep_tau_vs_e_energy_scale_barrel.count(period) )
         throw exception("Period '%1%'not found in tau vs ele map.") %period;
 
-    PhysicalValue e_fake_rate_correction = PhysicalValue(0.0,0.0);
+    StVariable e_fake_rate_correction = StVariable(0.0, 0.0, 0.0);
 
     if(tauVSeDiscriminator == TauIdDiscriminator::byDeepTau2017v2p1VSe){
 
-        if(std::abs(eta) < 1.448){
-            if(deep_tau_vs_e_energy_scale_eta_low.at(period).count(tauVSeWP))
-                e_fake_rate_correction = deep_tau_vs_e_energy_scale_eta_low.at(period).at(tauVSeWP);
+        if(std::abs(eta) < 1.460){
+            if(deep_tau_vs_e_energy_scale_barrel.at(period).count(decayMode))
+                e_fake_rate_correction = deep_tau_vs_e_energy_scale_barrel.at(period).at(decayMode);
             else
-                throw exception ("tau vs e wp (low eta map): '%1%' now allowed") %tauVSeWP ;
+                throw exception ("Decay mode for the fake e->tau (barrel): '%1%' now allowed") %decayMode ;
         }
         else if (std::abs(eta) > 1.558){
-            if(deep_tau_vs_e_energy_scale_eta_high.at(period).count(tauVSeWP))
-                e_fake_rate_correction = deep_tau_vs_e_energy_scale_eta_high.at(period).at(tauVSeWP);
+            if(deep_tau_vs_e_energy_scale_endcap.at(period).count(decayMode))
+                e_fake_rate_correction = deep_tau_vs_e_energy_scale_endcap.at(period).at(decayMode);
             else
-                throw exception ("tau vs e wp (high eta map): '%1%' now allowed") %tauVSeWP ;
+                throw exception ("Decay mode for the fake e->tau (endcap): '%1%' now allowed") %decayMode ;
         }
 
     }
-
-    auto e_fake_rate_final_correction = (e_fake_rate_correction.GetValue() +
-                                         static_cast<int>(scale) * e_fake_rate_correction.GetStatisticalError());
+    auto error = e_fake_rate_correction.decimals_to_print() > 0 ? e_fake_rate_correction.decimals_to_print() :
```

Comment:
- decimals_to_print is not an error.
- the value of the shift should depend on the sign of the scale